targets/psa_switch/psa_switch.cpp

```diff
@@ -337,17 +418,53 @@ PsaSwitch::ingress_thread() {
     ingress_mau->apply(packet.get());
     packet->reset_exit();
 
-    // prioritize dropping if marked as such - do not move below other checks
-    const auto &f_drop = phv->get_field("psa_ingress_output_metadata.drop");
-    if (f_drop.get_int()) {
+    // ingress cloning - sent in their pre-ingress-thread state to either an egress
+    //                   port or a multicast group
+    //                 - dropped packets should still be cloned - do not move below drop
+    unsigned int clone = phv->get_field("psa_ingress_output_metadata.clone").get_uint();
+    if (clone) {
+      MirroringSessionConfig config;
+      unsigned int clone_session_id = phv->get_field("psa_ingress_output_metadata.clone_session_id").get_uint();
+      bool is_session_configured = mirroring_get_session(
+          static_cast<mirror_id_t>(clone_session_id), &config);
+      BMLOG_DEBUG_PKT(*packet, "Cloning packet at ingress to session id {}", clone_session_id);
+
+      if (is_session_configured) {
+        const Packet::buffer_state_t packet_out_state = packet->save_buffer_state();
+        packet->restore_buffer_state(packet_in_state);
+
+        std::unique_ptr<Packet> packet_copy = packet->clone_no_phv_ptr();
+        packet_copy->set_register(PACKET_LENGTH_REG_IDX, ingress_packet_size);
+        auto phv_copy = packet_copy->get_phv();
+        phv_copy->reset_metadata();
+        phv_copy->get_field("psa_egress_parser_input_metadata.packet_path").set(PACKET_PATH_CLONE_I2E);
+        phv_copy->get_field("psa_egress_input_metadata.packet_path").set(PACKET_PATH_CLONE_I2E);
```

Comment:
This line is redundant and I'd recommend removing it, because egress_thread code always copies the value of psa_egress_parser_input_metadata.packet_path to psa_egress_input_metadata.packet_path for every packet, regardless of the value of packet_path.